xnmt/input.py

```diff
@@ -25,25 +26,38 @@ def get_padded_sent(self, token, pad_len):
 class SimpleSentenceInput(Input):
   """
   A simple sent, represented as a list of tokens
-  
+
   Args:
     words (List[int]): list of integer word ids
-    vocab (Vocab):
+    Annotation (Dict): annotation for the sentence
```

Comment:
Could you remove the annotation here, and instead directly pass the information to the component that needs it (maybe using ```!Ref```) if possible? Given the growing number of features in XNMT, I think it is becoming more and more important to keep the basic interface clean. (I think you mentioned before that this would be possible)